lib/libzpool/uzfs_mtree.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+#include <sys/dmu_objset.h>
+#include <sys/uzfs_zvol.h>
+#include <sys/dmu_traverse.h>
+#include <sys/dsl_pool.h>
+#include <sys/dsl_dataset.h>
+#include <sys/dsl_destroy.h>
+#include <sys/dmu_tx.h>
+
+#define	TXG_DIFF_SNAPNAME	"tsnap"
+
+struct diff_txg_blk {
+	avl_tree_t *tree;
+	uint64_t start_txg;
+	uint64_t end_txg;
+};
+
+void
+add_to_mblktree(avl_tree_t *tree, uint64_t boffset, uint64_t blen)
```

Comment:
why tree called as mblktree..? also, add comments at each logic.

Reply:
it contains modified block's details, so i kept it as mblktree. I will add comments at each tree operation/logic.

Reply:
add_to_txg_diff_tree